src/Service/ParcelShopsService.php

```diff
@@ -49,13 +49,21 @@ public function getParcelShopCollection(
 
         $parcelShopCollection = [];
         foreach ($rawParcelShopCollection as $rawParcelShop) {
+            if (isset($parcelShopCollection[$rawParcelShop->pclshopid])) {
```

Comment:
I know you are doing this in some context. But this should not be an library responsibility to check if id's are unique.